File: UninstallPackages.py
# -*- coding: utf-8 -*-
# @Time    : 2020/7/6 11:44
# @Author  : Jason
# @File    : main.py

# 批量卸载python包（卸载除pip和setuptools以外的所有包）
# 物理环境卸载方式：cmd中直接使用python运行该文件（python UninstallPackages.py）
# 虚拟环境卸载方式：1、cmd中进入相应的python虚拟环境,即命令行之前出现（yourvenv）
#                2、使用python运行该文件（python UninstallPackages.py）
# 其他卸载方式：1、IDE中运行该代码直接卸载当前环境的包

import os
packages = os.popen('pip list').read()
print('当前虚拟环境存在的包：\n'+packages)
packages = packages.split('\n')[2:-1]
for package in packages:
    package = package[0:package.find(' ')]
    if package != 'pip' and package != 'setuptools':
        print('\n即将卸载的包：'+package)
        os.system("@echo y | pip uninstall "+package)

File: UninstallPackages.rs
use std::process::Command;

fn main() {
    let mut output = Command::new("cmd")
        .arg("/C")
        .arg("pip")
        .arg("list")
        .output()
        .expect("failed to execute process");

    let result = String::from_utf8(output.stdout).expect("error");
    let mut lines: Vec<&str> = result.split('\n').skip(2).collect::<Vec<&str>>();
    lines.pop();
    for line in lines {
        if !(line.starts_with("pip") || line.starts_with("setuptools")) {
            println!("uninstall [{}]", line.trim());
            let pkg_name = line.split_whitespace().collect::<Vec<&str>>()[0];

            output = Command::new("cmd")
                .arg("/C")
                .arg("@echo")
                .arg("y")
                .arg("|")
                .arg("pip")
                .arg("uninstall")
                .arg(pkg_name)
                .output()
                .expect("failed to execute process");
            let result_out = String::from_utf8(output.stdout).expect("error");
            println!("{}", result_out)
        }
    }
    let mut tmp = String::new();
    let _ = std::io::stdin().read_line(&mut tmp);
}
